File: model/src/main/java/ru/mipt/acsl/decode/model/domain/proxy/DecodeProxyResolver.java
package ru.mipt.acsl.decode.model.domain.proxy;

import ru.mipt.acsl.decode.model.domain.DecodeReferenceable;
import ru.mipt.acsl.decode.model.domain.DecodeRegistry;
import org.jetbrains.annotations.NotNull;

import java.net.URI;

/**
 * @author Artem Shein
 */
public interface DecodeProxyResolver
{
    @NotNull
    <T extends DecodeReferenceable> DecodeResolvingResult<T> resolve(@NotNull DecodeRegistry registry, @NotNull URI uri,
                                                       @NotNull Class<T> cls);
}

File: model/src/main/scala/ru/mipt/acsl/decode/model/domain/proxy/DecodeProxyResolver.scala
package ru.mipt.acsl.decode.model.domain.proxy

import ru.mipt.acsl.decode.model.domain.Referenceable
import ru.mipt.acsl.decode.model.domain.proxy.path.ProxyPath
import ru.mipt.acsl.decode.model.domain.impl.registry.Registry

/**
  * @author Artem Shein
  */
trait DecodeProxyResolver {
  def resolveElement(registry: Registry, path: ProxyPath): (Option[Referenceable], ResolvingResult)
}
